docs/testing/unit-testing.md

```diff
@@ -0,0 +1,406 @@
+   npm test
+   ```
+
+## Examples
+
+The examples in this section use Jest with its default settings. These settings support ES5 JavaScript and CommonJS modules. See the [Jest documentation](https://jestjs.io/docs/getting-started) for how to configure Jest and node.js to support more recent versions of JavaScript, support TypeScript, and support ECMAScript modules. To run any of these examples, take the following steps:
+
+1. Create an Office add-in project for the appropriate Office host application (for example, Excel or Word). One way to do this quickly is to use the [Yo Office tool](https://github.com/OfficeDev/generator-office).
+1. In the root of the project, [install Jest](https://jestjs.io/docs/getting-started).
+1. [Install the office-addin-mock tool](#install-the-tool).
+1. Create a file exactly like the first file in the example and add it to the folder that contains the project's other source files, often called `\src`.
+1. Create a subfolder to the source file folder and give it an appropriate name, such as `\tests`.
+1. Create a file exactly like the test file in the example and add it to the subfolder.
+1. Add a `test` script to the package.json file, and then run the test, as described in [Basic usage](#basic-usage).
+
+### Mocking the Office Common APIs
+
+This example assumes an Office add-in for any host that supports the Office Common APIs (for example, Excel, PowerPoint, or Word). The add-in has one of its features in a file named `my-common-api-add-in-feature.js`. The following shows the contents of the file. The `addHelloWorldText` function sets the text "Hello World!" to whatever is currently selected in the document; for example; a range in Word, or a cell in Excel, or a text box in PowerPoint.
+
+```javascript
+const myCommonAPIAddinFeature = {
+
+    addHelloWorldText: async () => {
+        const options = { coercionType: Office.CoercionType.Text };
+        await Office.context.document.setSelectedDataAsync("Hello World!", options);
+    }
+}
+  
+module.exports = myCommonAPIAddinFeature;
+```
+
+The test file, named `my-common-api-add-in-feature.test.js` is in a subfolder, relative to the location of the add-in code file. The following shows the contents of the file. Note that the top level property is `context`, an [Office.Context](/javascript/api/office/office.context) object, so the object that is being mocked is the parent of this property: an [Office](/javascript/api/office) object. Note the following about this code:
+
+- The `OfficeMockObject` constructor does *not* add all of the Office enum classes to the mock `Office` object, so the `CoercionType.Text` value that is referenced in the add-in method must be added explicitly in the seed object.
+- Because the Office JavaScript APIs library isn't loaded in the node process, the `Office` object that is referenced in the add-in code must be declared and initialized.
+
+```javascript
+const OfficeMockObject = require("office-addin-mock");
```

Comment:
If you are using this `require` syntax, I think is it would be better to rename this variable to `OfficeAddinMock` in order for the constructor call later be: `OfficeAddinMock.OfficeMockObject` instead of  `OfficeMockObject.OfficeMockObject`. I think it will make the code cleaner to read.
This can also be applied to other parts of the article where this `require` call is made.